x-pack/plugins/security_solution/public/resolver/view/use_colors.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import euiThemeAmsterdamDark from '@elastic/eui/dist/eui_theme_amsterdam_dark.json';
+import euiThemeAmsterdamLight from '@elastic/eui/dist/eui_theme_amsterdam_light.json';
+import { useMemo } from 'react';
+import { useUiSetting } from '../../../../../../src/plugins/kibana_react/public';
+
+type ResolverColorNames =
+  | 'descriptionText'
+  | 'full'
+  | 'graphControls'
+  | 'graphControlsBackground'
+  | 'resolverBackground'
+  | 'resolverEdge'
+  | 'resolverEdgeText'
+  | 'resolverBreadcrumbBackground'
+  | 'pillStroke'
+  | 'triggerBackingFill'
+  | 'processBackingFill';
+type ColorMap = Record<ResolverColorNames, string>;
+
+/**
+ * Get access to Kibana-theme based colors.
+ */
+export function useColors(): ColorMap {
+  const isDarkMode = useUiSetting('theme:darkMode');
+  const theme = isDarkMode ? euiThemeAmsterdamDark : euiThemeAmsterdamLight;
+  return useMemo(() => {
+    const darkOverrides = {
+      processBackingFill: `${theme.euiColorPrimary}1F`,
+      resolverEdge: theme.euiColorLightShade,
+      resolverEdgeText: theme.euiColorFullShade,
+      triggerBackingFill: `${theme.euiColorDanger}1F`,
+    };
+
+    return {
+      descriptionText: theme.euiColorDarkestShade,
+      full: theme.euiColorFullShade,
+      graphControls: theme.euiColorDarkestShade,
+      graphControlsBackground: theme.euiColorEmptyShade,
+      processBackingFill: `${theme.euiColorPrimary}0F`, // Add opacity 0F = 6% , 1F = 12%
+      resolverBackground: theme.euiColorEmptyShade,
+      resolverEdge: theme.euiColorLightestShade,
+      resolverBreadcrumbBackground: theme.euiColorLightestShade,
+      resolverEdgeText: theme.euiColorDarkShade,
+      triggerBackingFill: `${theme.euiColorDanger}0F`,
+      pillStroke: theme.euiColorLightShade,
+      ...(isDarkMode ? darkOverrides : {}),
```

Comment:
:grey_question: Just personal preference, but I found the `getThemedOptions` a little easier to follow. Does doing it this way offer better perf?

Reply:
nope, i just thought `getThemedOptions` was confusing. What if we go back to an approach similar to `getThemedOptions` but we use a ternary instead?

```
triggerBackingFill: `${theme.euiColorDanger}${isDarkMode ? : '1F': '0F'}`
```

Reply:
That would do the trick, just so if I'm editing colors/fills I don't have a sneaky "other place" I could forget to update.

Reply:
@bkimmel hows this?
```
/**
 * Get access to Kibana-theme based colors.
 */
export function useColors(): ColorMap {
  const isDarkMode = useUiSetting('theme:darkMode');
  const theme = isDarkMode ? euiThemeAmsterdamDark : euiThemeAmsterdamLight;
  return useMemo(() => {
    return {
      descriptionText: theme.euiColorDarkestShade,
      full: theme.euiColorFullShade,
      graphControls: theme.euiColorDarkestShade,
      graphControlsBackground: theme.euiColorEmptyShade,
      processBackingFill: `${theme.euiColorPrimary}${isDarkMode ? '1F' : '0F'}`, // Add opacity 0F = 6% , 1F = 12%
      resolverBackground: theme.euiColorEmptyShade,
      resolverEdge: isDarkMode ? theme.euiColorLightShade : theme.euiColorLightestShade,
      resolverBreadcrumbBackground: theme.euiColorLightestShade,
      resolverEdgeText: isDarkMode ? theme.euiColorFullShade : theme.euiColorDarkShade,
      triggerBackingFill: `${theme.euiColorDanger}${isDarkMode ? '1F' : '0F'}`,
      pillStroke: theme.euiColorLightShade,
    };
  }, [isDarkMode, theme]);
}
```

Reply:
that's great: need to update a color, do it in one place.